// slowMath.add(6, 2).then(num => {
//     console.log(num);
//     return slowMath.multiply(num, 2);
// }).then(num => {
//     console.log(num);
//     return slowMath.divide(num,4);
// }).then(num => {
//     console.log(num);
//     return slowMath.subtract(num, 3);
// }).then(num => {
//     console.log(num)
//     return slowMath.add(num, 98);
// }).then(num => {
//     console.log(num)
// return slowMath.divide(num,2);
// }).then(num => {
//     console.log(num)
//     return slowMath.multiply(num, 50);
// }).then(num => {
//     console.log(num)
//     return slowMath.divide(num,40);
// }).then(num =>{
//     console.log(num)
//     return slowMath.add(num, 32);
// }).then(num => {
//     console.log(`The final result is ${num}`)
// }).catch(err => {
//     console.log(err)
// });

////Async/Await
let doMath = (slowMath)

try {
    let message = slowMath.add(6, 2) 
    if (message === 4) {(num => {
    console.log(num);
    } else {let err = new Error('calculating error');
    reject(err);
}return slowMath.multiply(num, 2);
// })try(num => {
//     console.log(num);
//     return slowMath.divide(num,4);
// })try(num => {
//     console.log(num);
//     return slowMath.subtract(num, 3);
// })try(num => {
//     console.log(num)
//     return slowMath.add(num, 98);
// })try(num => {
//     console.log(num)
// return slowMath.divide(num,2);
// })try(num => {
//     console.log(num)
//     return slowMath.multiply(num, 50);
// })try(num => {
//     console.log(num)
//     return slowMath.divide(num,40);
// })try(num =>{
//     console.log(num)
//     return slowMath.add(num, 32);
// })try(num => {
//     console.log(`The final result is ${num}`)
// }).catch(err => {
//     console.log(err)
// });
